auction_keeper/main.py

```diff
@@ -356,21 +356,47 @@ def auction_handled_by_this_shard(self, id: int) -> bool:
             logging.debug(f"Auction {id} is not handled by shard {self.arguments.shard_id}")
             return False
 
+    def can_bite(self, ilk: Ilk, urn: Urn, box: Rad, dunk: Rad, chop: Wad) -> bool:
+        # Typechecking intentionally omitted to improve performance
+        rate = ilk.rate
+
+        # Collateral value should be less than the product of our stablecoin debt and the debt multiplier
+        safe = Ray(urn.ink) * ilk.spot >= Ray(urn.art) * rate
+        if safe:
+            return False
+
+        # Ensure there's room in the litter box
+        litter = self.cat.litter()
+        room: Rad = box - litter
+        if litter >= box:
+            return False
+        if room < ilk.dust:
+            return False
+
+        # Prevent null auction (ilk.dunk [Rad], ilk.rate [Ray], ilk.chop [Wad])
+        dart: Wad = min(urn.art, Wad(min(dunk, room) / Rad(ilk.rate) / Rad(chop)))
+        dink: Wad = min(urn.ink, urn.ink * dart / urn.art)
+        return dart > Wad(0) and dink > Wad(0)
+
     def check_vaults(self):
         started = datetime.now()
-        ilk = self.vat.ilk(self.ilk.name)
-        rate = ilk.rate
         available_dai = self.mcd.dai.balance_of(self.our_address) + Wad(self.vat.dai(self.our_address))
+        box = self.cat.box()
+        dunk = self.cat.dunk(self.ilk)
+        chop = self.cat.chop(self.ilk)
 
         # Look for unsafe vaults and bite them
         urns = self.urn_history.get_urns()
         logging.debug(f"Evaluating {len(urns)} {self.ilk} urns to be bitten if any are unsafe")
 
-        for urn in urns.values():
-            if self.is_shutting_down():
-                return
+        for i, urn in enumerate(urns.values()):
+            if i % 500 == 0:  # Every 500 vaults, free some CPU and then update ilk.rate
```

Comment:
Do you think it will be good to change this value from config/keeper arguments?

Reply:
The number of configuration parameters has already been cited as a complexity risk.  I'd like to retire a few before considering adding any more parameters.  Removing params is expensive because we must consider impact on `dockerized-auction-keeper`.